### ISSUE TITLE
Issue with coverting NetworkX graph into FNSS

### ISSUE THREAD

**Issue by Ali00** [fnss/fnss#29]
I have a json topology file, which I successfully converted into a networkx graph (G). I have faced an issue when converting the resulted graph G into a fnss object. The error message says:
((**File "build/bdist.linux-i686/egg/fnss/adapters/autonetkit.py", line 58, in from_autonetkit
  File "build/bdist.linux-i686/egg/fnss/topologies/topology.py", line 528, in rename_edge_attribute**))

**Here is my python scrip:**
```
from __future__ import print_function
import networkx as nx
from networkx.readwrite import json_graph
import json
from itertools import tee, izip
import collections
from collections import OrderedDict
import fnss
def pairwise(iterable):
    a, b = tee(iterable)
    next(b, None)
    return izip(a, b) 
def Path_ports(path = [], *args):
    global G
    x = 0
    output_ports=[]
    for pair in pairwise(path):
        port_info = G[pair[0]][pair[1]]["ports"]
        print (port_info)
        for key,value in port_info.items():
            if key == path[x]:
               print (key)
               print(value)
               output_ports.append(value)
        x +=1
    return (output_ports)

#----------------------------------------------------
def To_fnss():
    global G
    topology = fnss.from_autonetkit(G)
    ns=[]
    es=[]
    ns= fnss.Topology.nodes(G, data=False)
    es= fnss.Topology.edges(G, nbunch=None, data=False)
    print ("The number of nodes in this topology is : "), len(ns)
    print ("The number of edges in this topology is : "), len(es)
#------------------------------------------------------
gnl = json.load(open("MyTopo.json"))
print("gnl = {}".format(gnl))
g = json_graph.node_link_graph(gnl)
G = nx.Graph()
G = g.copy()
print("Network nodes: {}".format(g.nodes()))
print("Network links: {}".format(g.edges()))
print("Network nodes: {}".format(g.nodes(data=True)))
print("Network links: {}".format(g.edges(data=True)))
for n in g.nodes():
    node_type = g.node[n].get("type", None)
    vid_info = ""
    node_vid = g.node[n].get("vnid", None)
    if node_vid:
        vid_info = "with vid = {}".format(node_vid)
    print("Node {} is a {} {}".format(n, node_type, vid_info))
    if node_type == "host":
       print ("Hi this is a Host")
       G.remove_node(n)
for link in g.edges():
    src = link[0]
    trg = link[1]
    port_info = g[src][trg]["ports"]
    cap = g[src][trg]["capacity"]
    print("link from {} port {} to {} port {} has capacity {}"
          .format(src, port_info[src], trg, port_info[trg], cap))
print (g.nodes())
print (G.nodes())
src = "S1"
dst = "S6"
path = nx.shortest_path(G, source= src, target = dst)
print (path)
Ports = Path_ports(path)
print (Ports)
To_fnss()
```

![issue](https://user-images.githubusercontent.com/12594727/110860259-7266dc80-82b4-11eb-89fc-9ca9cb4d4780.png)

**Here is the json file topology:**
```
{
    "directed": false,
    "multigraph": false,
    "graph": {
        "name": "My Topology",
        "desc": "Example"
    },
    "nodes": [
        {
            "ip": "10.0.0.1",
            "mac": "00:00:00:00:00:1",
            "y": 358,
            "x": 456,
            "type": "host",
            "id": "H1"
        },
        {
            "ip": "10.0.0.2",
            "mac": "00:00:00:00:00:2",
            "y": 211,
            "x": 607,
            "type": "host",
            "id": "H2"
        },
        {
            "y": 142,
            "x": 209,
            "type": "switch",
            "id": "S1"
        },
        {
            "y": 157,
            "x": 593,
            "type": "switch",
            "id": "S2"
        },
        {
            "y": 323,
            "x": 419,
            "type": "switch",
            "id": "S3"
        },
        {
            "y": 107,
            "x": 408,
            "type": "switch",
            "id": "S4"
        },
        {
            "y": 247,
            "x": 230,
            "type": "switch",
            "id": "S5"
        },
        {
            "y": 247,
            "x": 230,
            "type": "switch",
            "id": "S6"
        }
    ],
    "links": [
        {
            "source": "H1",
            "capacity": 40,
            "weight": 2,
            "target": "S1",
            "ports": {
                "H1": 1,
                "S1": 1
            }
        },
        {
            "source": "H2",
            "capacity": 40,
            "weight": 2,
            "target": "S6",
            "ports": {
                "H2": 1,
                "S6": 1
            }
        },
        {
            "source": "S1",
            "capacity": 11,
            "weight": 20,
            "target": "S2",
            "ports": {
                "S1": 2,
                "S2": 1
            }
        },
        {
            "source": "S1",
            "capacity": 20,
            "weight": 7,
            "target": "S4",
            "ports": {
                "S1": 3,
                "S4": 1
            }
        },
        {
            "source": "S2",
            "capacity": 40,
            "weight": 2,
            "target": "S3",
            "ports": {
                "S2": 2,
                "S3": 1
            }
        },
        {
            "source": "S4",
            "capacity": 40,
            "weight": 2,
            "target": "S5",
            "ports": {
                "S4": 2,
                "S5": 1
            }
        },
        {
            "source": "S3",
            "capacity": 40,
            "weight": 2,
            "target": "S5",
            "ports": {
                "S3": 2,
                "S5": 2
            }
        },
        {
            "source": "S3",
            "capacity": 16,
            "weight": 10,
            "target": "S6",
            "ports": {
                "S3": 3,
                "S6": 2
            }
        }
    ]
}
```

**Comment by lorenzosaino** [comment]
Hi Ali, could you please edit your comment so that the code is included into code blocks with correct indentation? It's hard to follow the code. Reference: https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks

**Comment by lorenzosaino** [comment]
Can you also report what versions of Python, FNSS and networkx are you using?

**Comment by Ali00** [comment]
> Hi Ali, could you please edit your comment so that the code is included into code blocks with correct indentation? It's hard to follow the code. Reference: https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks

Hi Lorenzo, 
Sorry about that, I have updated the comment.

**Comment by Ali00** [comment]
> Can you also report what versions of Python, FNSS and networkx are you using?

Python 2.7.12
NetworkX 2.2
FNSS 0.8

**Comment by lorenzosaino** [comment]
Hi Ali, I looked into it. 

Are you sure you are running FNSS 0.8.0? I cannot reproduce your issue with FNSS 0.8.0. All references to `edges_iter` (like the one in your error message) were removed by https://github.com/fnss/fnss/commit/aad3c6b9b1cd582eeffb0cfbdd43e7287818e899, which is included in the 0.8.0 release. In fact, line 528 of `topology.py` does not reference `edges_iter`. See: https://github.com/fnss/fnss/blob/v0.8.0/fnss/topologies/topology.py#L528. That suggests you may be running an earlier version of FNSS. If you use FNSS 0.8.0 or later, it should be working fine.

I am closing this ticket for now, feel free to reopen it if you find further issues.

As a side note, Python 2.7 went into end of life more than one year ago and fnss will drop support for it soon. fnss 0.8 is almost 3 years old and doesn't support versions of networkx more recent than 2.2. You might want to consider using Python 3 and fnss 0.9.1 to be able to use more recent versions of networkx.
